### PR TITLE
[SEDONA-109] Add ST_OrderingEquals to Apache Sedona

### PR DESCRIPTION
**Pull request by madzik555**
## Did you read the Contributor Guide?
- Yes, I have read [Contributor Rules](https://sedona.apache.org/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/community/develop/)

## Is this PR related to a JIRA ticket?
- Yes, the URL of the assoicated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-109 The PR name follows the format `[SEDONA-109] my subject`.

## What changes were proposed in this PR?
ST_OrderingEquals function

## How was this patch tested?
Scala, Python integration tests

## Did this PR include necessary documentation updates?
- Yes, I have updated the documentation update.

### CODE REVIEW

**Review by jiayuasu — CHANGES_REQUESTED** [review]
Thanks for this PR. Please move OrderingEquals to Predicate.

**Comment by madzik555** [comment]
Should I rewrite the function code to src/main/java/org/apache/sedona/flink/expressions/Predicates.java? Or should I only change the docs?

**Comment by jiayuasu** [comment]
> Should I rewrite the function code to src/main/java/org/apache/sedona/flink/expressions/Predicates.java

@madzik555 Yes, please move the code to Predicates.java

File: docs/api/flink/Predicate.md [hunk 1]
```diff
@@ -41,4 +41,25 @@ Spark SQL example:
...
+```
+
+## ST_OrderingEquals
+Introduction: Returns true if the geometries are equal and the coordinates are in the same order
+
+Format: `ST_OrderingEquals(A: geometry, B: geometry)`
+
+Since: `v1.2.1`
+
+Spark SQL example 1:
```
Comment (jiayuasu) [inline docs/api/flink/Predicate.md:53]: Please remove `spark` here because this is for Flink.

File: python/tests/sql/test_predicate.py [hunk 2]
```diff
@@ -206,3 +206,13 @@ def test_st_overlaps(self):
         not_overlaps = self.spark.sql("select ST_Overlaps(c,d) from testtable")
         assert overlaps.take(1)[0][0]
         assert not not_overlaps.take(1)[0][0]
+
+    def test_st_ordering_equals_ok(self):
```
Comment (jiayuasu) [inline python/tests/sql/test_predicate.py:210]: Can you add a similar test for Spark SQL in Scala?

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: python/tests/sql/test_predicate.py (hunk 2 above)
Comment (madzik555) [reply python/tests/sql/test_predicate.py:210]: Yes :)

File: docs/api/flink/Predicate.md (hunk 1 above)
Comment (jiayuasu) [reply docs/api/flink/Predicate.md:53]: You removed the `spark` from the wrong place. You didn't remove this.

**Review by jiayuasu — CHANGES_REQUESTED** [review]

**Comment by madzik555** [comment]
Hey, @jiayuasu could you re-run checks?
Error in Scala and Java build (2.4.8, 2.11.8):
`Failed to execute goal on project sedona-flink_2.11: Could not resolve dependencies for project org.apache.sedona:sedona-flink_2.11:jar:1.2.1-incubating-SNAPSHOT: Failed to collect dependencies at org.apache.flink:flink-streaming-java_2.11:jar:1.14.3: Failed to read artifact descriptor for org.apache.flink:flink-streaming-java_2.11:jar:1.14.3: Could not transfer artifact org.apache.flink:flink-streaming-java_2.11:pom:1.14.3 from/to central (https://repo.maven.apache.org/maven2): transfer failed for https://repo.maven.apache.org/maven2/org/apache/flink/flink-streaming-java_2.11/1.14.3/flink-streaming-java_2.11-1.14.3.pom: Connection timed out (Read failed) -> [Help 1]`

**Comment by jiayuasu** [comment]
@madzik555 Hi Magdalena, I’d like to send you some information regarding Sedona project management. If it is ok, could you please send me an email so I can know your email address? My email is jiayu@apache.org   Thank you!

**Pull request merged**